src/client/csfle/state_machine.rs

```diff
@@ -181,6 +209,77 @@ impl CryptExecutor {
     }
 }
 
+#[derive(Debug)]
+struct Mongocryptd {
+    opts: MongocryptdOptions,
+    client: Client,
+    child: Mutex<Result<Process>>,
+}
+
+impl Mongocryptd {
+    async fn new(opts: MongocryptdOptions) -> Result<Self> {
+        let child = Mutex::new(Ok(Self::spawn(&opts)?));
+        let uri = opts.uri.as_deref().unwrap_or("mongodb://localhost:27020");
```

Comment:
nit: could we make this and the server selection timeout named constants on this type

Reply:
Done.